src/core/sio1-server.cc

```diff
@@ -57,5 +66,54 @@ void PCSX::SIO1Server::startServer(uv_loop_t* loop, int port) {
 
 void PCSX::SIO1Server::stopServer() {
     m_serverStatus = SIO1ServerStatus::SERVER_STOPPING;
+    g_emulator->m_counters->m_pollSIO1 = false;
     m_fifoListener.stop();
 }
+
+PCSX::SIO1Client::SIO1Client() : m_listener(g_system->m_eventBus) {
+    m_listener.listen<Events::SettingsLoaded>([this](const auto& event) {
+        if (g_emulator->settings.get<Emulator::SettingDebugSettings>().get<Emulator::DebugSettings::SIO1Client>() &&
+            (m_clientStatus != SIO1ClientStatus::CLIENT_STARTED)) {
+                startClient(std::string_view(g_emulator->settings.get<Emulator::SettingDebugSettings>().get<Emulator::DebugSettings::SIO1ClientHost>().value),
+                        g_emulator->settings.get<Emulator::SettingDebugSettings>().get<Emulator::DebugSettings::SIO1ClientPort>());
+        }
+    });
+    m_listener.listen<Events::Quitting>([this](const auto& event) {
+        if (m_clientStatus == SIO1ClientStatus::CLIENT_STARTED) stopClient();
+    });
+}
+
+void PCSX::SIO1Client::startClient(std::string_view address, unsigned port) {
+    if (m_clientStatus == SIO1ClientStatus::CLIENT_STARTED) {
+        throw std::runtime_error("Client already started");
+    }
+    auto& emuSettings = PCSX::g_emulator->settings;
+    auto& debugSettings = emuSettings.get<Emulator::SettingDebugSettings>();
+    auto SIO1ModeSettings = debugSettings.get<Emulator::DebugSettings::SIO1ModeSetting>().value;
+    if (SIO1ModeSettings == Emulator::DebugSettings::SIO1Mode::Raw) {
+        g_emulator->m_sio1->m_sio1Mode = SIO1::SIO1Mode::Raw;
+    } else {
+        g_emulator->m_sio1->m_sio1Mode = SIO1::SIO1Mode::Protobuf;
+    }
+
+    m_clientStatus = SIO1ClientStatus::CLIENT_STARTED;
+    g_emulator->m_sio1->m_fifo.setFile(new UvFifo(address, port));
+    if (g_emulator->m_sio1->fifoError()) {
+        m_clientStatus = SIO1ClientStatus::CLIENT_STOPPING;
+        g_emulator->m_counters->m_pollSIO1 = false;
+        stopClient();
+    }
+
+    g_system->printf("%s", _("SIO1 client connecting...\n"));
+    if (!g_emulator->m_sio1->m_fifo.asA<UvFifo>()->isConnecting())
```

Comment:
I removed this for now, and will look into a callback for the near future.

Reply:
There is a check inside of sio1.cc that simply polls the isConnecting state, as long as the fifo is healthy, it will report connected to the user. Safer because it doesn't race, as it's in a spot that if you weren't connected, you would never take the path.